docs/2.4/admin-guide.md

```diff
@@ -0,0 +1,1479 @@
+```
+
+### Recording Proxy Mode
+
+See [Audit Log Architecture](architecture/#audit-log) to understand how the session
+recording works. The important detail is that by default the recording is not
+available if a cluster runs `sshd`, the OpenSSH server, on the nodes.
+
+To enable session recording for `sshd` nodes, the cluster must be switched to
+"recording proxy" mode, in this mode the recording will be done on the proxy level:
+
+```bash
+# snippet from /etc/teleport.yaml
+auth_service:
+   session_recording: "proxy"  # can also be "off" and "on" (default)
+```
+
+Next, `sshd` must be told to allow users to log in with certificates generated by the
+Teleport User CA. Start by exporting the Teleport CA public key:
+
+```bash
+$ tctl auth export --type=user
+```
+
+To allow access to a single user, copy the above output to `~/.ssh/authorized_keys`, but
+to apply this for all users, remove "cert-authority" from the start of `tctl` output and
+copy it to `/etc/ssh/teleport_user_ca.pub`. 
+
+Add the following line to `/etc/ssh/sshd_config`: 
+
+```bash
+TrustedUserCAKeys /etc/ssh/teleport_user_ca.pub
+```
+
+Now `sshd` will trust users who present a Teleport-issued certificate. Next step 
+is to configure host authentication.
+
+When in recording mode, Teleport will check that the host certificate of the
+node a user connects to is signed by a Teleport CA. By default this is a strict
+check, if the node presents just a key, or a certificate signed by a different
+CA, Teleport will reject this connection with the error message saying 
+_"ssh: handshake failed: remote host presented a public key, expected a host certificate"_
+
+You can disable strict host checks as shown below. Note: this opens the possibility for 
+Man-in-the-Middle (MITM) attacks and is not recommended.
+
+```bash
+# snippet from /etc/teleport.yaml
+auth_service:
+  proxy_checks_host_keys: no
+```
+
+The best solution is to ask Teleport to issue valid host certificates for all OpenSSH 
+nodes. To generate a host certificate run this on your auth server:
+
+```bash
+$ tctl auth sign \
+      --host=node.example.com \
+      --format=openssh
+```
+
+Then add the following lines to `/etc/ssh/sshd_config` and restart sshd.
+
+```bash
+HostKey /etc/ssh/teleport_host_key
+HostCertificate /etc/ssh/teleport_host_key-cert.pub
+```
+
+Now you can use `tsh ssh user@host.example.com` to login into any `sshd` node in the cluster
+and the session will be recorded. But if you want to use OpenSSH `ssh` client for logging
+into `sshd` servers behind a proxy in "recording mode", you have to tell the `ssh` client
+to use the jump host and enable the agent forwarding (otherwise a recording proxy will not 
+be able to terminate the SSH connection to record it):
+
+```bash
+# Note that agent forwarding is enabled twice: one from a client to a proxy
+# (mandatory if using a recording proxy), and then # from a proxy to the end
+# server (optional)
```

Comment:
`and then optionally from a proxy to the end server if you want your agent running on the end server or not`.